assignment5.py

```diff
@@ -2,9 +2,71 @@
 from utilityModule import *
 
 
+def get_user_ratings(dataframe):
+    user_ratings_dict = {}
+    for index, row in dataframe.iterrows():
+        user_id = row["userId"]
+        movie_id = row["movieId"]
+        rating = row["rating"]
+        if user_id not in user_ratings_dict:
+            user_ratings_dict[user_id] = {}
+        user_ratings_dict[user_id][movie_id] = rating
+    return user_ratings_dict
+
+
+# Calculate the similarity between two users based on their movie ratings
+def calculate_similarity(user1, user2, us_ratings):
+    if user1 not in us_ratings or user2 not in us_ratings:
+        return 0
+    movies1 = set(user_ratings[user1].keys())
+    movies2 = set(user_ratings[user2].keys())
+    common_movies = movies1.intersection(movies2)
+    if len(common_movies) == 0:
+        return 0
+    common_movies = set(us_ratings[user1]).intersection(set(us_ratings[user2]))
+    if len(common_movies) == 0:
+        return 0
+    user1_ratings = [us_ratings[user1][movie] for movie in common_movies]
+    user2_ratings = [us_ratings[user2][movie] for movie in common_movies]
+    num = sum([a * b for a, b in zip(user1_ratings, user2_ratings)])
+    denom1 = sum([rating ** 2 for rating in user1_ratings])
+    denom2 = sum([rating ** 2 for rating in user2_ratings])
+    denom = (denom1 ** 0.5) * (denom2 ** 0.5)
+    if denom == 0:
+        return 0
+    return num / denom
```

Comment:
I guess you could have used a function from a couple assignments back (I think it was no. 3), but this looks ok